WrightTools/kit/_array.py

```diff
@@ -237,7 +237,15 @@ def share_nans(*arrs) -> tuple:
     list
         List of nD arrays in same order as given, with nan indicies syncronized.
     """
-    nans = np.zeros(joint_shape(*arrs))
+    typ = int
```

Comment:
Consider simpler syntax e.g.
```
typ = np.complex128 if "c" in [arr.dtype.kind for arr in arrs] else arrs[0].dtype
```

Note that `np.nan * complex(0) = nan + nanj`, so no concern here about 0-valued imaginary parts

Reply:
what happens if the 2nd array is float and 1st is int?   I was considering having to broadcast into the most "np.nan"-like of all datatypes, so the hierarchy was int < float < complex   and that any float would require an int to broadcast to float.

Reply:
That's a fair point.  I will note that none of these array creation code blocks you have revised really think about integer types (`dtype.kind`=`u`,`i`).  Consideration of integer types might be worth looking into as a separate issue?  e.g. quite a few more tests should be implemented to really tackle the int type...

Reply:
The good news is that casting ints as floats doesn't have dramatic consequences for our typical usage.   I can't think of an example where this would really be a problem...

Reply:
somehow the original code was able to handle it, yet I couldn't understand how...a simple line of code would be nicer though

Reply:
To keep this hierarchy and make the code more readable, consider an elif structure:
```
kinds = {arr.dtype.kinds for arr in arrs}
if "c" in kinds:
    dtype = complex
elif "f" in kinds:
   dtype = float
elif "i" in kinds:
   dtype = int
else:
   dtype = arrs[0].dtype
```